SwiftUIHW/SwiftUIHW/VPNView/VPNPageView.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  ContentView.swift
+//  SwiftUIHW
+//
+//  Created by Daniil on 16.01.2023.
+//
+
+import SwiftUI
+
+/// Страница подключения к впн
+struct VPNPageView: View {
+    // MARK: - Public properties
+    
+    var body: some View {
+        NavigationView {
+            ZStack {
+                Color.black.ignoresSafeArea()
+                VStack(spacing: Constants.firstVStackSpacing) {
+                    
+                    VStack(spacing: Constants.secondVStackSpacing) {
+                        countyImageView
+                        turnOnToggleView
+                    }
+                    
+                    VStack(spacing: Constants.thirdVStackSpacing) {
+                        vpnInfoView(parameter: Constants.IPAddressText, value: Constants.IPAddressValue)
+                        vpnInfoView(parameter: Constants.receivedText, value: "\(viewModel.receiverValue) \(Constants.MBText)")
+                        vpnInfoView(parameter: Constants.sentText, value: "\(viewModel.sentValue) \(Constants.MBText)")
+                    }
+                    .padding(.bottom, Constants.fortyPaddingValue)
+                    navigationLinkView.padding(.bottom, Constants.thirtyPaddingValue)
+                }
+            }
+        }
+    }
+    
+    // MARK: - Private properties
+    
+    @StateObject private var countriesList = CountriesList()
+    @StateObject private var viewModel = VPNViewModel()
+    
+    private var countyImageView: some View {
+        Image(countriesList.imageName)
+            .resizable()
+            .frame(width: Constants.frameWidthValue, height: Constants.frameHeightValue)
+            .cornerRadius(Constants.cornerRadiusValue)
+    }
+    
+    private var turnOnToggleView: some View {
+        Toggle(isOn: $viewModel.isVpnOn) {
+                EmptyView()
+        }
+        .onChange(of: viewModel.isVpnOn, perform: { _ in
+            viewModel.isVpnOn ? viewModel.dataUpdate() : viewModel.makeDataDefaultValue()
```

Comment:
```suggestion
            viewModel.dataUpdate()
```
   а внутри уже логика сама, что вызвать в зависимости от viewModel.isVpnOn